### PR TITLE
New dashboard: remove legacy dashboard

### LINKED ISSUE #12085: Tests: remove `test` checks
We have deprecated the old dashboard now, so we should remove the `test` check since we are already covering all the new tests with `test-ext-theme`

### LINKED ISSUE #12086: Templates: remove old dashboard files
We have deprecated the old dashboard. We are ready to remove all the code related to it now. There HTML, CSS and JS files. Besides, there are some Invoke tasks to _build_ these assets that can be removed as well.

### PR DESCRIPTION
**Pull request by humitos**
- Closes #12085 
- Closes #12086

Related https://github.com/readthedocs/readthedocs-ops/issues/1600

### CODE REVIEW

File: readthedocs/projects/forms.py [hunk 1]
```diff
@@ -479,9 +479,8 @@ def __init__(self, *args, **kwargs):
                 self.fields.pop(field)
 
         # Remove analytics from new dashboard
-        if settings.RTD_EXT_THEME_ENABLED:
-            for field in ["analytics_code", "analytics_disabled"]:
-                self.fields.pop(field)
+        for field in ["analytics_code", "analytics_disabled"]:
+            self.fields.pop(field)
```
Comment (stsewd) [inline readthedocs/projects/forms.py:483]: Looks like we can just don't define these attributes since they are always removed.

File: readthedocs/projects/urls/private.py [hunk 2]
```diff
@@ -250,17 +242,15 @@
...
-    addons_urls = [
-        re_path(
-            r"^(?P<project_slug>[-\w]+)/addons/edit/$$",
-            AddonsConfigUpdate.as_view(),
-            name="projects_addons",
-        ),
-    ]
+addons_urls = [
+    re_path(
+        r"^(?P<project_slug>[-\w]+)/addons/edit/$$",
```
Comment (stsewd) [inline readthedocs/projects/urls/private.py:247]: This was probably working just by chance?

```suggestion
        r"^(?P<project_slug>[-\w]+)/addons/edit/$",
```

File: readthedocs/rtd_tests/tests/test_views.py [hunk 3]
```diff
@@ -186,16 +186,6 @@ def setUp(self):
         self.pip.save()
         self.pip.versions.update(privacy_level=PUBLIC)
 
-    @mock.patch("readthedocs.projects.tasks.builds.update_docs_task")
-    def test_build_redirect(self, mock):
```
Comment (stsewd) [inline readthedocs/rtd_tests/tests/test_views.py:190]: doesn't this still redirects somewhere else?

File: readthedocs/rtd_tests/tests/test_views.py [hunk 4]
```diff
@@ -213,71 +203,6 @@ def test_build_list_includes_external_versions(self):
         self.assertEqual(response.status_code, 200)
         self.assertIn(external_version_build, response.context["build_qs"])
 
-    @mock.patch("readthedocs.projects.tasks.builds.update_docs_task")
-    @mock.patch("readthedocs.core.utils.cancel_build")
-    def test_rebuild_specific_commit(self, mock_update_docs_task, mock_cancel_build):
```
Comment (stsewd) [inline readthedocs/rtd_tests/tests/test_views.py:218]: All these deleted tests don't look to be specific to the old theme?

File: readthedocs/settings/base.py [hunk 5]
```diff
@@ -311,24 +299,21 @@ def INSTALLED_APPS(self):  # noqa
             apps.append("readthedocsext.cdn")
             apps.append("readthedocsext.donate")
             apps.append("readthedocsext.spamfighting")
-        if self.RTD_EXT_THEME_ENABLED:
-            apps.append("readthedocsext.theme")
         if self.SHOW_DEBUG_TOOLBAR:
             apps.append("debug_toolbar")
 
+        if ext_theme:
+            apps.append("readthedocsext.theme")
```
Comment (stsewd) [inline readthedocs/settings/base.py:306]: Should probably always include this, as without exttheme the app would no longer work

File: readthedocs/settings/base.py [hunk 6]
```diff
@@ -430,7 +414,8 @@ def MIDDLEWARE(self):
     @property
     def TEMPLATES(self):
         dirs = [self.TEMPLATE_ROOT]
-        if self.RTD_EXT_THEME_ENABLED:
+
+        if ext_theme:
```
Comment (stsewd) [inline readthedocs/settings/base.py:418]: same here, this should always be executed

File: readthedocs/rtd_tests/tests/test_views.py (hunk 4 above)
Comment (humitos) [reply readthedocs/rtd_tests/tests/test_views.py:218]: These tests are for the POST request done to the dashboard to rebuild. That view was deleted and is now done via APIv3. That's why I deleted these tests as well.

File: readthedocs/settings/base.py (hunk 5 above)
Comment (humitos) [reply readthedocs/settings/base.py:306]: I tried that, but I had to add this conditional because there are some scenarios where exttheme is not present: embed testings, for example.

We can work on polishing that in another PR if we want; but I didn't want to block this work due to that.

**Comment by humitos** [comment]
I did a test for this at https://readthedocs.slack.com/archives/G81T0N8S3/p1745509009820219 and it worked as expected. I think we are ready to merge this PR. We can fix any edge case that I may be missing as we go.

File: readthedocs/projects/views/private.py [hunk 7]
```diff
@@ -102,40 +102,35 @@ def get_context_data(self, **kwargs):
...
-            n_projects = projects.count()
-
-            # TODO remove this with RTD_EXT_THEME_ENABLED
-            # This is going to try hard to show the new dashboard announcement.
-            # We can't yet back down to another announcement as we don't have
-            # the ability to evaluate local storage. Until we add the ability to
-            # dynamically change the announcement, this is going to be the only
-            # announcement shown.
-            if True:  # pylint: disable=using-constant-test
-                template_name = "new-dashboard.html"
```
Comment (agjohnson) [inline readthedocs/projects/views/private.py:122]: Ahh yes this is the onboarding announcement I was describing above :+1:

**Review by agjohnson — APPROVED** [review]
This looks great :+1: 

The only thing that makes me a little uneasy is the static file removal. I'm not certain about some of those files. I know we shouldn't be referencing them with any projects built rather recently, but I do worry they are referenced by builds that we aren't considering.

So I might suggest we feed some of the important files in that list into Cloudflare to see if we have any volume of hits on files from referred sources that seem legitimate.

File: readthedocs/core/notifications.py [hunk 8]
```diff
@@ -37,21 +37,12 @@
...
         ).strip(),
         body=textwrap.dedent(
             """
-            {% if RTD_EXT_THEME_ENABLED %}
-              We are beginning to direct users to our new dashboard as we work to retire our legacy dashboard.
-            {% else %}
-              You are currently using our legacy dashboard, which will be retired on <time datetime="2025-03-11">March 11th, 2025</time>.
-              You should <a href="//{{ SWITCH_PRODUCTION_DOMAIN }}{% url "account_login" %}">switch to our new dashboard</a> before then.
-            {% endif %}
+            We are beginning to direct users to our new dashboard as we work to retire our legacy dashboard.
```
[the remark this answers is not in the document]
Comment (alparslan-uv) [reply readthedocs/core/notifications.py:45]: Ok
